core/module/refinements_spec.rb

```diff
@@ -0,0 +1,22 @@
+require_relative '../../spec_helper'
+
+describe "Module#refinements" do
+  ruby_version_is "3.2" do
+    it "returns list of classes redefined in module" do
+      refined_integer = nil
+      refined_string = nil
+
+      m = Module.new do
+        refine Integer do
+          refined_integer = self
+        end
+
+        refine String do
+          refined_string = self
+        end
+      end
+
+      m.refinements.should == [refined_integer, refined_string]
```

Comment:
**minor:** Not sure whether the order is guarantied. I suppose it isn't, so we may just sort both arrays, I mean:
```suggestion
      m.refinements.sort.should == [refined_integer, refined_string].sort
```

Reply:
Hm, right. Will sorting by `object_id` or `to_s` be working?

Reply:
Yes.

And in `used_refinements` (https://github.com/ruby/spec/pull/1028) also order is not guarantied, so sorting probably also makes sense.